src/Sylius/Bundle/ShippingBundle/Form/Type/ShippingMethodChoiceType.php

```diff
@@ -82,10 +81,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
     public function configureOptions(OptionsResolver $resolver)
     {
         $choiceList = function (Options $options) {
+            $methods = $methods = $this->repository->findAll();
+
             if (isset($options['subject'])) {
-                $methods = $this->resolver->getSupportedMethods($options['subject'], $options['criteria']);
-            } else {
-                $methods = $this->repository->findBy($options['criteria']);
+                /** @var MethodsResolverInterface $resolver */
+                foreach ($this->resolversRegistry->all() as $resolver) {
+                    if ($resolver->supports($options['subject'])) {
+                        $methods = $resolver->getSupportedMethods($options['subject']);
+                        break;
+                    }
+                }
```

Comment:
Let's move this logic to the "main" methods resolver and inject it here. It will support every shipment and use exactly this logic to get the right methods. (CompositeMethodsResolver as name, wdyt?)